src/stuf/stuf_zds/tests/test_client.py

```diff
@@ -155,3 +156,13 @@ def test_non_latin1_characters(self):
                 self.fail("Body encoding should succeed")
             except RegistrationFailed:
                 pass
+
+    @tag("gh-4146")
+    def test_custom_timeout_applied(self):
+        stuf_service = StufServiceFactory.build(
+            soap_service__url="https://httpstat.us/", soap_service__timeout=1
+        )
+        client = StufZDSClient(stuf_service, {})  # type: ignore
+
+        with self.assertRaises(Timeout):
+            client.get("https://httpstat.us/200?sleep=3000")
```

Comment:
Can't we test this with a mocked request and not a real one?

Reply:
yes, see the test in ape-pie: https://github.com/maykinmedia/ape-pie/blob/979972db0797e0c0b50aa14db801c7e8e9ac170f/tests/test_config_adapter.py#L56

You must always be able to run the test suite offline.

Reply:
I got my inspiration from this test:
https://github.com/open-formulieren/open-forms/blob/264e0644f88efe5734889677700cf86df538b7fa/src/soap/tests/test_client.py#L246-L259

Probably it will need to be updated

Reply:
I vaguely remember there being a very specific edge cases that required this approach due to the zeep abstractions